### ISSUE TITLE
Startcom Certificate

### ISSUE THREAD

**Issue by bengadbois** [PrivateBin/privatebin.info-pelican#11]
Not an issue with the code, but https://privatebin.info/ is using a StartCom certificate, which starting with Chrome 56+ gets flagged as insecure/broken HTTPS

https://security.googleblog.com/2016/10/distrusting-wosign-and-startcom.html

**Comment by elrido** [comment]
It is correct that we are using a StartCom cert for all the privatebin domains, but ours was issued before October 21st, so Chrome still accepts it. I just verified that (again) on Chrome 56.

**Comment by rugk** [comment]
> Beginning with Chrome 56, certificates issued by WoSign and StartCom after October 21, 2016 00:00:00 UTC will not be trusted. Certificates issued before this date may continue to be trusted, for a time, if they comply with the Certificate Transparency in Chrome policy or are issued to a limited set of domains known to be customers of WoSign and StartCom.

We don't use certificate transparency so this works be a first thing we have to do.
And of course we can forget to get in such an exception list by chrome.

But generally:
> In subsequent Chrome releases, these exceptions will be reduced and ultimately removed, culminating in the full distrust of these CAs.

So maybe chrome 57, 58 or so?? I'd rather not let that happen.

> This staged approach is solely to ensure sites have the opportunity to transition to other Certificate Authorities that are still trusted in Google Chrome,

So we should do …😊

Also we are playing with the fire:
> Any attempt by WoSign or StartCom to circumvent these controls will result in immediate and complete removal of trust.

Finally we need to lookup what's Firefox way.

So all in all I would appreciate a fast switch. @elrido does not want to use let's encrypt, for reasons I do not quite understand, but it would certainly be the easiest way.
But something has to happen, that's clear. And rather sooner than later, when it may be too late.

**Comment by elrido** [comment]
The reason is simple and I explained it to you in great: lets encrypt by default requires me to install a software with root access on the server that messes with the configuration behind the back of the configuration management and orchestration solution I have already in place. The logical way around that is to run lets encrypt locally and deploy the generated certs via orchestration. This is handled using a git repository so I can easily track all versions of my configuration (and certificates are part of that). But the short lifetime of these lets encrypt certs means that I have to run this multiple times per year. That increases my maintenance workload. I rather pay some CA a few bucks every few years if it saves me that work.

Bottom line: You either pay with your time or your money. Or you don't care about security and give some software root access to your certificates and webservers configuration and content in a way that isn't easy to audit after the fact.

Edit for the german speaking: [Here is blog post](https://blog.fefe.de/?ts=a89f4ed6) with an opinion piece on why one might not wan't to install the let's encrypt script on your server.

**Comment by bengadbois** [comment]
I'm on Chrome 59.0.3033.0 (Official Build) Canary and it's not trusted right now

![distrusted](http://i.imgur.com/ZcrQ96q.png)

**Comment by elrido** [comment]
I found [another, much simpler tool](https://github.com/diafygi/acme-tiny) to handle the Let's encrypt challenges that doesn't require root user privileges or private key access and that I could integrate into my orchestration solution to automate the whole process. Hope that finally puts this to rest.

**Comment by rugk** [comment]
Yep, see Scott Helme's guides for this, it is really useful.

**Comment by elrido** [comment]
Scotts guide actually has a few insecure points: He still runs the script on the server in question and even allows the dedicated script user to sudo to control the webserver (restarting, reloading, etc.).

I prefer to run that script on a dedicated instance with its own webserver for statically providing the acme_challenges and hook that path up into the actual instance via a reverse proxy rule. The generated certificates get picked up and distributed by the orchestration which also handles reloading of the production webserver, if needed.
